src/routes/safe/store/middleware/notificationsMiddleware.js

```diff
@@ -18,9 +18,42 @@ import { ADD_TRANSACTIONS } from '~/routes/safe/store/actions/addTransactions'
 import updateSafe from '~/routes/safe/store/actions/updateSafe'
 import { safeParamAddressFromStateSelector, safesMapSelector } from '~/routes/safe/store/selectors'
 import { type GlobalState } from '~/store/'
+import { loadFromStorage, saveToStorage } from '~/utils/storage'
 
 const watchedActions = [ADD_TRANSACTIONS, ADD_INCOMING_TRANSACTIONS, ADD_SAFE]
 
+const sendAwaitingTransactionNotification = async (
+  dispatch: Function,
+  safeAddress: string,
+  awaitingTxsSubmissionDateList: List[],
+  notificationKey: string,
+  notificationClickedCb: Function,
+) => {
+  const LAST_TIME_USED_LOGGED_IN_ID = 'LAST_TIME_USED_LOGGED_IN_ID'
+  if (!dispatch || !safeAddress || !awaitingTxsSubmissionDateList || !notificationKey) {
+    return
+  }
+  if (awaitingTxsSubmissionDateList.size === 0) {
+    return
+  }
+
+  let lastTimeUserLoggedIn = (await loadFromStorage(LAST_TIME_USED_LOGGED_IN_ID)) || null
+
+  const filteredDuplicatedAwaitingTxList = awaitingTxsSubmissionDateList.filter(submissionDate => {
+    return lastTimeUserLoggedIn ? submissionDate > lastTimeUserLoggedIn : true
```

Comment:
I'm not sure if this works `submissionDate > lastTimeUserLoggedIn`. It compares the strings and not actual dates, doesn't it?

Reply:
You are right, now should be ok